pyDeltaRCM/shared_tools.py

```diff
@@ -193,7 +193,7 @@ def get_weight_at_cell(ind, stage_nbrs, depth_nbrs, ct_nbrs, stage, qx, qy,
         weight = weight / np.nansum(weight)
         weight[nanWeight] = 0
     else:
-        weight[~nanWeight] = 1 / len(weight[~nanWeight])
+        weight[~nanWeight] = 1 / np.maximum(1, len(weight[~nanWeight]))
         weight[nanWeight] = 0
```

Comment:
amazing when what I'm sure was a few hours of cumulative effort boils down to a 15-character single-line change.

Your way of fixing is probably faster than the if-else tree I set up, and I see why taking the `maximum` prevents the div-by-zero error, but I just want to check that the new behavior is as expected.

If I understand correctly, if going into the if-else `weight = array([nan, nan, nan, nan, nan, nan, nan, nan, nan])`, then coming out `weight = array([0, 0, 0, 0, 0, 0, 0, 0, 0])`. 

Would this be problematic for anything? The expected behavior on the way the old maintainers wrote it suggests that `weight` should always come back with something.  E.g., `weight = array([0, 0, nan, nan, nan, nan, nan, nan, nan])`would become `weight = array([0.5, 0.5, 0, 0, 0, 0, 0, 0, 0])`. 

I'm guessing that we only started to have this problem based on the change Eric made to compute the weights for every cell only once at the top of the water parcel iteration, rather than with every step of the water parcel, because this change made it so that we actually calculate weights of "land" cells at all. To this end, I begun to look at just skipping the weight computation altogether for cells with type==land (returning some dummy weight), but never explored it really. 

I guess that your fix would probably work all the same though, and it probably doesn't make much difference for computational speed. 

Anyway, I'm approving these changes, but just wanted to run my ideas by you before merge.

Reply:
> I'm guessing that we only started to have this problem based on the change Eric made to compute the weights for every cell only once at the top of the water parcel iteration, rather than with every step of the water parcel, because this change made it so that we actually calculate weights of "land" cells at all.

This is my take on it as well. I think that the singular computation of weights that we are now doing across the entire array (even across -2 'land' cells) has saved us a lot of time, but a consequence was creating this edge case for the weights. So I agree with you, and yes the output would be `weight = array([0, 0, 0, 0, 0, 0, 0, 0, 0])` - this would only present problems if there was another bug allowing parcels to route into "invalid" cells where they'd encounter the fully 0 weight arrays. 

> I guess that your fix would probably work all the same though, and it probably doesn't make much difference for computational speed.

Yeah I thought about this a bit and figured that the land area is typically a small portion of the domain, so I couldn't imagine it was a big deal.

Reply:
cool I agree with all of this. Merging now.